packages/frontend/src/providers/DashboardProvider.tsx

```diff
@@ -429,6 +434,19 @@ export const DashboardProvider: React.FC = ({ children }) => {
         [],
     );
 
+    const addResultsCacheTime = useCallback((cacheMetadata: CacheMetadata) => {
+        if (cacheMetadata.cacheHit && cacheMetadata.cacheUpdatedTime) {
+            setResultsCacheTimes((old) => [
+                ...old,
+                cacheMetadata.cacheUpdatedTime,
+            ]);
+        }
+    }, []);
+
+    const getOldestCacheTime = useCallback(() => {
+        return min(resultsCacheTimes);
+    }, [resultsCacheTimes]);
```

Comment:
What is the advantage of returning a callback instead of the oldest cache time value?

Reply:
None really as is. I often like getters as a pattern cause they are flexible to add to, but it's not doing anything here. I can make it a value if you prefer.